src/dune_rules/merlin/merlin.ml

```diff
@@ -171,8 +171,10 @@ module Processed = struct
         :: flags
     in
     let suffixes =
-      List.map extensions ~f:(fun { Ml_kind.Dict.impl; intf } ->
-          make_directive "SUFFIX" (Sexp.Atom (Printf.sprintf "%s %s" impl intf)))
+      List.map extensions ~f:(fun { Ml_kind.Dialect.impl; intf } ->
+          make_directive "SUFFIX"
+            (Sexp.Atom
+               (Printf.sprintf "%s %s" impl (Option.value intf ~default:""))))
```

Comment:
I'm not sure what is consuming that sexp (merlin I presume) but it needs to be able to parse this correctly.

Sidenote: putting a space in atom like this does not look correct (even if there's no problem as a csexp). If we fix one issue we should fix the other one by making a dict of extensions.

Reply:
If this is consumed by [this](https://github.com/ocaml/merlin/blob/29ad7e1cbcfcbd9eb94ab22fb7a632cf384004d3/src/kernel/mconfig_dot.ml#L70) maybe we should just write `~default:impl`

Reply:
done